nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/service/ColdStorageService.java

```diff
@@ -0,0 +1,136 @@
+
+    /**
+     * Restores the cold content associated with the document of the given {@link DocumentRef} into its main storage.
+     * <p/>
+     * The permission {@value @value org.nuxeo.coldstorage.helpers.ColdStorageHelper#WRITE_COLD_STORAGE} is required.
+     *
+     * @implSpec This method will rely on the {@link org.nuxeo.ecm.core.blob.BlobProvider#getStatus(ManagedBlob)} to
+     *           check if the restore can be done, otherwise it will request a retrieval
+     *           {@link #requestRetrievalFromColdStorage(CoreSession, DocumentRef, Duration)}
+     * @return the updated document model if the restore succeeds
+     * @throws NuxeoException if the cold content is already in the main storage, if there is no cold content associated
+     *             with the given document, or if the user does not have the permissions needed to perform the action.
+     * @since 10.10
+     */
+    public DocumentModel restoreContentFromColdStorage(CoreSession session, DocumentRef documentRef);
+
+    public DocumentModel restoreFromColdStorage(CoreSession session, DocumentRef documentRef);
+
+    /**
+     * Restores the main content from ColdStorage.
+     *
+     * @implSpec: fires a {@value org.nuxeo.coldstorage.ColdStorageConstants#COLD_STORAGE_CONTENT_TO_RESTORE_EVENT_NAME}
+     *            event.
+     * @see #restoreContentFromColdStorage(CoreSession, DocumentRef)
+     */
+    void restoreMainContent(DocumentModel documentModel);
+
+    /**
+     * Move to ColdStorage all duplicated documents with the same content.
+     *
+     * @param session the session
+     * @param documentModel the document model
+     */
+    public void moveDuplicatedBlobToColdStorage(CoreSession session, DocumentModel documentModel);
+
+    /**
+     * Checks if the retrieved cold storage contents are available for download.
+     *
+     * @implSpec: Queries all documents with a cold storage content which are being retrieved, meaning
+     *            {@value org.nuxeo.coldstorage.ColdStorageConstants#COLD_STORAGE_BEING_RETRIEVED_PROPERTY} is
+     *            {@code true}, and it checks if it is available for download. In which case its fires a
+     *            {@value org.nuxeo.coldstorage.ColdStorageConstants#COLD_STORAGE_CONTENT_AVAILABLE_EVENT_NAME} event.
+     * @see #requestRetrievalFromColdStorage(CoreSession, DocumentRef, Duration)
+     */
+    public ColdStorageContentStatus checkColdStorageContentAvailability(CoreSession session);
+
+    /**
+     * Gets the number of the days where the document's blob is available, once it's retrieved from cold storage.
+     *
+     * @return the number of days of availability if property
+     *         {@value org.nuxeo.coldstorage.ColdStorageConstants#COLD_STORAGE_NUMBER_OF_DAYS_OF_AVAILABILITY_PROPERTY_NAME}
+     *         is configured, {@code 1} otherwise.
+     * @since 10.10
+     */
+    public int getNumberOfDaysOfAvailability();
```

Comment:
Could it be interesting to have such method instead?
```
Duration getDurationAvailability();
```

Reply:
Yes it make sense